python/ray/exceptions.py

```diff
@@ -130,8 +130,19 @@ def __init__(
         self.traceback_str = traceback_str
         self.actor_repr = actor_repr
         self._actor_id = actor_id
-        # TODO(edoakes): should we handle non-serializable exception objects?
         self.cause = cause
+
+        # If the cause is not serializable, it will cause worker crash.
+        # We need to catch it early, log a warning, and avoid it.
+        try:
+            pickle.dumps(self.cause)
+        except Exception as e:
+            err_msg = f"Unable to serialize exception ({self.cause.__class__}): {e}"
```

Comment:
`The original cause of RayTaskError ({self.cause.__class__}) is non-serializable: {e}. Overwriting the cause to be RayError` ?